File: cpp/test/Ice/operations/AllTests.cpp
// **********************************************************************
//
// Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <Ice/Ice.h>
#include <Ice/Locator.h>
#include <TestCommon.h>
#include <Test.h>

using namespace std;

Test::MyClassPrx
allTests(const Ice::CommunicatorPtr& communicator, bool collocated)
{
    string ref = "test:default -p 12010";
    Ice::ObjectPrx base = communicator->stringToProxy(ref);
    Test::MyClassPrx cl = Test::MyClassPrx::checkedCast(base);
    Test::MyDerivedClassPrx derived = Test::MyDerivedClassPrx::checkedCast(cl);

    cout << "testing twoway operations... " << flush;
    void twoways(const Ice::CommunicatorPtr&, const Test::MyClassPrx&);
    twoways(communicator, cl);
    twoways(communicator, derived);
    derived->opDerived();
    cout << "ok" << endl;

    cout << "testing oneway operations... " << flush;
    void oneways(const Ice::CommunicatorPtr&, const Test::MyClassPrx&);
    oneways(communicator, cl);
    cout << "ok" << endl;

    if(!collocated)
    {
        cout << "testing twoway operations with AMI... " << flush;
        void twowaysAMI(const Ice::CommunicatorPtr&, const Test::MyClassPrx&);
        twowaysAMI(communicator, cl);
        twowaysAMI(communicator, derived);
        cout << "ok" << endl;

        cout << "testing twoway operations with new AMI mapping... " << flush;
        void twowaysNewAMI(const Ice::CommunicatorPtr&, const Test::MyClassPrx&);
        twowaysNewAMI(communicator, cl);
        twowaysNewAMI(communicator, derived);
        cout << "ok" << endl;

        cout << "testing oneway operations with AMI... " << flush;
        void onewaysAMI(const Ice::CommunicatorPtr&, const Test::MyClassPrx&);
        onewaysAMI(communicator, cl);
        cout << "ok" << endl;

        cout << "testing oneway operations with new AMI mapping... " << flush;
        void onewaysNewAMI(const Ice::CommunicatorPtr&, const Test::MyClassPrx&);
        onewaysNewAMI(communicator, cl);
        cout << "ok" << endl;

        cout << "testing batch oneway operations... " << flush;
        void batchOneways(const Test::MyClassPrx&);
        batchOneways(cl);
        batchOneways(derived);
        cout << "ok" << endl;
    }

    return cl;
}

File: py/test/Ice/operations/AllTests.py
# **********************************************************************
#
# Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
#
# This copy of Ice is licensed to you under the terms described in the
# ICE_LICENSE file included in this distribution.
#
# **********************************************************************

import Ice, Test, Twoways, TwowaysAMI, TwowaysNewAMI, Oneways, OnewaysAMI, OnewaysNewAMI, BatchOneways, sys

def test(b):
    if not b:
        raise RuntimeError('test assertion failed')

def allTests(communicator, collocated):
    ref = "test:default -p 12010"
    base = communicator.stringToProxy(ref)
    cl = Test.MyClassPrx.checkedCast(base)
    derived = Test.MyDerivedClassPrx.checkedCast(cl)

    sys.stdout.write("testing twoway operations... ")
    sys.stdout.flush()
    Twoways.twoways(communicator, cl)
    Twoways.twoways(communicator, derived)
    derived.opDerived()
    print("ok")

    sys.stdout.write("testing oneway operations... ")
    sys.stdout.flush()
    Oneways.oneways(communicator, cl)
    print("ok")

    if not collocated:
        sys.stdout.write("testing twoway operations with AMI... ")
        sys.stdout.flush()
        TwowaysAMI.twowaysAMI(communicator, cl)
        print("ok")

        sys.stdout.write("testing twoway operations with new AMI mapping... ")
        sys.stdout.flush()
        TwowaysNewAMI.twowaysNewAMI(communicator, cl)
        print("ok")

        sys.stdout.write("testing oneway operations with AMI... ")
        sys.stdout.flush()
        OnewaysAMI.onewaysAMI(communicator, cl)
        print("ok")

        sys.stdout.write("testing oneway operations with new AMI mapping... ")
        sys.stdout.flush()
        OnewaysNewAMI.onewaysNewAMI(communicator, cl)
        print("ok")

        sys.stdout.write("testing batch oneway operations...  ")
        sys.stdout.flush()
        BatchOneways.batchOneways(cl)
        BatchOneways.batchOneways(derived)
        print("ok")

    return cl
